engine/gameobject/src/gameobject/res_anim.cpp

```diff
@@ -1,26 +1,33 @@
 // Copyright 2020 The Defold Foundation
 // Licensed under the Defold License version 1.0 (the "License"); you may not use
 // this file except in compliance with the License.
-// 
+//
 // You may obtain a copy of the License, together with FAQs at
 // https://www.defold.com/license
-// 
+//
 // Unless required by applicable law or agreed to in writing, software distributed
 // under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 // CONDITIONS OF ANY KIND, either express or implied. See the License for the
 // specific language governing permissions and limitations under the License.
 
-#include "res_anim.h"
+#include <dmsdk/resource/resource.h>
 
 namespace dmGameObject
 {
-    dmResource::Result ResAnimCreate(const dmResource::ResourceCreateParams& params)
+    static dmResource::Result ResAnimCreate(const dmResource::ResourceCreateParams& params)
     {
         return dmResource::RESULT_NOT_SUPPORTED;
     }
 
-    dmResource::Result ResAnimDestroy(const dmResource::ResourceDestroyParams& params)
+    static dmResource::Result ResAnimDestroy(const dmResource::ResourceDestroyParams& params)
     {
         return dmResource::RESULT_NOT_SUPPORTED;
     }
+
+    static dmResource::Result RegisterResourceTypeAnim(dmResource::ResourceTypeRegisterContext& ctx)
+    {
+        return dmResource::RegisterType(ctx.m_Factory, ctx.m_Name,0,0,ResAnimCreate,0,ResAnimDestroy,0);
+    }
 }
+
+DM_DECLARE_RESOURCE_TYPE(ResourceTypeAnim, "animc", dmGameObject::RegisterResourceTypeAnim, 0);
```

Comment:
This one _is_ a bit weird, since it's empty.
I believe it's because the component types require an associated resource type to go with it.